dockstore-integration-testing/src/test/java/io/dockstore/client/cli/BasicIT.java

```diff
@@ -1346,4 +1362,63 @@ public void testRefreshingUserMetadata() {
         final long count = testingPostgres.runSelectStatement("select count(*) from user_profile where location='Toronto'", long.class);
         Assert.assertEquals("One user should have this info now, there are " + count, 1, count);
     }
+
+    @Test
+    public void eventResourcePaginationTest() {
+        ApiClient client = getWebClient(USER_1_USERNAME, testingPostgres);
+        ContainersApi toolsApi = new ContainersApi(client);
+        ContainertagsApi toolTagsApi = new ContainertagsApi(client);
+
+        DockstoreTool tool = manualRegisterAndPublish(toolsApi, "dockstoretestuser", "dockerhubandgithub", "regular",
+                "git@github.com:DockstoreTestUser/dockstore-whalesay.git", "/Dockstore.cwl", "/Dockstore.wdl", "/Dockerfile",
+                DockstoreTool.RegistryEnum.DOCKER_HUB, "master", "latest", true);
+        EventsApi eventsApi = new EventsApi(client);
+        List<Event> events = eventsApi.getEvents(EventSearchType.STARRED_ENTRIES.toString(), 10, 0);
+        Assert.assertEquals("No starred entries, so there should be no events returned", 0, events.size());
+        StarRequest starRequest = new StarRequest();
+        starRequest.setStar(true);
+        toolsApi.starEntry(tool.getId(), starRequest);
+        events = eventsApi.getEvents(EventSearchType.STARRED_ENTRIES.toString(), 10, 0);
+        Assert.assertEquals("Should be an event for the tag that was automatically created for the newly registered tool", 1, events.size());
+        // Add and update tag 101 times
+        Set<String> randomTagNames = new HashSet<>();
+
+        for (int i = 0; i < 110; i++) {
+            randomTagNames.add(RandomStringUtils.randomAlphanumeric(255));
+        }
+        randomTagNames.forEach(randomTagName -> {
+            List<Tag> randomTags = getRandomTags(randomTagName);
+            toolTagsApi.addTags(tool.getId(), randomTags);
+        });
+        try {
+            events = eventsApi.getEvents(EventSearchType.STARRED_ENTRIES.toString(), 101, 0);
+            Assert.fail("Should've failed because it's over the limit");
+        } catch (ApiException e) {
+            Assert.assertEquals("{\"errors\":[\"query param limit must be less than or equal to 100\"]}", e.getMessage());
```

Comment:
This is bound to change over time, make both this and the web resource access the same constant and parameterize the `100`